test-util/android-test/src/main/java/com/pronoidsoftware/testutil/androidtest/core/data/di/TestSessionStorageModule.kt

```diff
@@ -0,0 +1,34 @@
+package com.pronoidsoftware.testutil.androidtest.core.data.di
+
+import android.content.SharedPreferences
+import com.pronoidsoftware.core.data.auth.EncryptedSessionStorage
+import com.pronoidsoftware.core.data.di.SessionStorageModule
+import com.pronoidsoftware.core.domain.AuthInfo
+import com.pronoidsoftware.core.domain.SessionStorage
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.components.SingletonComponent
+import dagger.hilt.testing.TestInstallIn
+import javax.inject.Singleton
+import kotlinx.coroutines.runBlocking
+
+@Module
+@TestInstallIn(
+    components = [SingletonComponent::class],
+    replaces = [SessionStorageModule::class],
+)
+object TestSessionStorageModule {
+
+    @Provides
+    @Singleton
+    fun provideTestSessionStorage(
+        sharedPreferences: SharedPreferences,
+        authInfo: AuthInfo?,
+    ): SessionStorage {
+        val sessionStorage = EncryptedSessionStorage(sharedPreferences)
+        runBlocking {
+            sessionStorage.set(authInfo)
+        }
+        return sessionStorage
+    }
```

Comment:
I assume the auth info object is statically provided here, how would you handle the scenario where you would not want to have an initial session? Could it make sense maybe leaving the session empty by default and creating a JUnit extension that automatically sets and unsets it for specific tests?

Reply:
I looked into this today and unfortunately could not find a better way... I wish I could use a JUnit rule to make it cleaner, but I could not figure out how to properly manage hilt injection other than uninstalling the auth info module and installing a replacement like how I do in `AuthFeatureE2ETest` in this PR. I keep the auth info provided as the stub since the majority of the test will start with a sign in state and then uninstall it and provide null in tests where I need to start from a logged out state. 

The problem seems to arise because I need the auth info saved in the session storage at time of app launch in the test so that it properly navigates to the agenda feature. I want to simulate real app behavior as much as possible and avoid having to sign in on each test. In order to do so, it seems that dependency injection with hilt is the only way to provide it in time. I tried considering HiltAndroidRule's `delayComponentReady` API perhaps, but then couldn't seem to programmatically set if the auth info should be null or not for injection. And hilt's `@TestInstallIn` annotation forces overriding of the module it replaces, which I don't want to happen every time. So for now, I think this is the best approach, although I don't think it's particularly elegant...

Open to suggestions or to discussing further, but will keep this as is for now since it's working as expected, just perhaps with more code than truly is necessary 😅